include/zephyr/net/ping.h

```diff
@@ -0,0 +1,78 @@
+/*
```

Comment:
I would like to see in the commit message some reasoning why we would need this API.
Especially as we have currently quite well working ping implementation in net-shell. Perhaps this code here could be combined with the "net ping" command. For example so that "net ping" could call public API implemented in this PR.

What is also missing in current ICMP handlers, is a possibility to register user data to it. Meaning that when the handler is called, some opaque user data could be passed to the handler function. That part was unfortunately missing when I created the original ICMP handler API.

Reply:
The primary reason for even adding this is that in many chips that use "socket offload" style drivers, there is a function for ping, but they lack the ability to use raw ICMP. As such, these chips could not be made to work with the existing ping implementation despite it working well in most cases with other chips. This is a stripped down interface for ping so that such chips can perform ping operations for testing. I do however agree that it might be a good choice to integrate into the existing net shell, and in fact it is using the ICMP logic from there in the case where native drivers are used.